roles/confluent.common/tasks/remove_packages.yml

```diff
@@ -0,0 +1,49 @@
+---
+- name: Get Package Facts
+  package_facts:
+
+- name: Determine if Confluent Platform Package Version Will Change
+  set_fact:
+    confluent_package_version_changed: >-
+      {{ ansible_facts.packages['confluent-common'] is defined
+          and ansible_facts.packages['confluent-common'][0].version
+            not in [confluent_package_version, confluent_full_package_version] }}
+
+- name: Get installed Confluent Packages
+  set_fact:
+    confluent_packages_actual: >
+      {{ ansible_facts.packages | dict2items | map(attribute='key') | select('match', 'confluent-') | list }}
+
+- name: Determine Confluent Packages to Remove
+  set_fact:
+    confluent_packages_removed: >
+      {{ confluent_packages_actual if confluent_package_version_changed|bool else
+          confluent_packages_actual | intersect(['confluent-kafka']) if confluent_server_enabled else
```

Comment:
missing confluent-kafka-2.12 case

Reply:
would it be 
```
confluent_packages_actual | intersect(['confluent-kafka', 'confluent-kafka-2.12']) if confluent_server_enabled else
```

Reply:
This is really great work btw, great use of the built in filters

Reply:
I do not think it is required, as the `confluent-kafka-2.12` would be removed by a version change, removing all `confluent-` packages. But I can add it, if it makes things clearer. 😄

Reply:
Nope you are correct. Thanks for explaining!